web/src/main/java/org/springframework/security/web/savedrequest/CookieRequestCache.java

```diff
@@ -113,11 +113,11 @@ public void removeRequest(HttpServletRequest request, HttpServletResponse respon
 		response.addCookie(removeSavedRequestCookie);
 	}
 
-	private static String encodeCookie(String cookieValue) {
+	protected String encodeCookie(String cookieValue) {
```

Comment:
I think it would be more powerful to introduce `Converter`s that can turn a `Cookie` into a `String` and vice-versa. Something like:

```java
public void setCookieValueDecoder(Converter<HttpServletRequest, String> decoder);

public void setValueCookieEncoder(Converter<String, Cookie> encoder);
```

In addition to allowing you to customize how the value is encoded and decoded, this approach makes more features available as well.

I specifically like the encoder since this allows an application to completely construct the cookie, allowing for things like setting the same site parameter or customizing the max age. This contract would also allow the application to customize the cookie name.

On the flip side, the decoder could look up the cookie and extract the URI themselves. This contract allows the application to customize the cookie name.